mm2src/mm2_main/src/lp_healthcheck.rs

```diff
@@ -0,0 +1,277 @@
+            );
+            return false;
+        }
+
+        let Ok(public_key) = Libp2pPublic::try_decode_protobuf(&self.data.sender_public_key) else {
+            log::debug!("Couldn't decode public key from the healthcheck message.");
+
+            return false
+        };
+
+        if self.data.sender_peer != public_key.to_peer_id().to_string() {
+            log::debug!("`sender_peer` and `sender_public_key` doesn't belong each other.");
+
+            return false;
+        }
+
+        let Ok(encoded_message) = self.data.encode() else {
+            log::debug!("Couldn't encode healthcheck data.");
+            return false
+        };
+
+        let res = public_key.verify(&encoded_message, &self.signature);
+
+        if !res {
+            log::debug!("Healthcheck isn't signed correctly.");
+        }
+
+        res
+    }
+
+    #[inline]
+    pub(crate) fn encode(&self) -> Result<Vec<u8>, rmp_serde::encode::Error> { encode_message(self) }
+
+    #[inline]
+    pub(crate) fn decode(bytes: &[u8]) -> Result<Self, rmp_serde::decode::Error> { decode_message(bytes) }
+
+    #[inline]
+    pub(crate) fn should_reply(&self) -> bool { !self.data.is_a_reply }
+
+    #[inline]
+    pub(crate) fn sender_peer(&self) -> &str { &self.data.sender_peer }
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+#[cfg_attr(any(test, target_arch = "wasm32"), derive(PartialEq))]
+struct HealthcheckData {
+    sender_peer: String,
+    sender_public_key: Vec<u8>,
+    target_peer: String,
+    expires_at: i64,
+    is_a_reply: bool,
+}
+
+impl HealthcheckData {
+    #[inline]
+    fn encode(&self) -> Result<Vec<u8>, rmp_serde::encode::Error> { encode_message(self) }
+}
+
+#[inline]
+pub fn peer_healthcheck_topic(peer_id: &PeerId) -> String {
+    pub_sub_topic(PEER_HEALTHCHECK_PREFIX, &peer_id.to_string())
+}
+
+#[derive(Deserialize)]
+pub struct RequestPayload {
+    peer_id: String,
+}
+
+#[derive(Debug, Display, Serialize, SerializeErrorType)]
+#[serde(tag = "error_type", content = "error_data")]
+pub enum HealthcheckRpcError {
+    InvalidPeerAddress { reason: String },
+    MessageGenerationFailed { reason: String },
+    MessageEncodingFailed { reason: String },
+}
+
+impl HttpStatusCode for HealthcheckRpcError {
+    fn status_code(&self) -> common::StatusCode {
+        match self {
+            HealthcheckRpcError::InvalidPeerAddress { .. } => StatusCode::BAD_REQUEST,
+            HealthcheckRpcError::MessageGenerationFailed { .. } | HealthcheckRpcError::MessageEncodingFailed { .. } => {
+                StatusCode::INTERNAL_SERVER_ERROR
+            },
+        }
+    }
+}
+
+pub async fn peer_connection_healthcheck_rpc(
+    ctx: MmArc,
+    req: RequestPayload,
+) -> Result<bool, MmError<HealthcheckRpcError>> {
+    /// When things go awry, we want records to clear themselves to keep the memory clean of unused data.
+    /// This is unrelated to the timeout logic.
+    const ADDRESS_RECORD_EXPIRATION: Duration = Duration::from_secs(60);
```

Comment:
why not use `ctx.healthcheck_config.timeout_ms` as well for this task.

they both refer to the same timeout anyway, it's just that this is the sender end and the other is the receiver (they share the same oneshot).